pytest/tests/mocknet/load_testing.py

```diff
@@ -0,0 +1,45 @@
+# Force each node to submit many transactions for
+# about 20 minutes. Monitor the block production time
+# stays consistent.
+
+import sys, time
+from rc import pmap
+
+from load_testing_helper import TIMEOUT
+
+sys.path.append('lib')
+import mocknet
+
+nodes = mocknet.get_nodes()
+
+
+def check_stats(include_tps=False):
+    stats = mocknet.measure_chain_stats(nodes[-1])
+    (block_time, stdev) = stats['block_time']
+    assert block_time < 1.01
+    assert stdev < 0.1
+    if include_tps:
+        (tps, tps_stdev) = stats['tps']
+        assert tps > 190
```

Comment:
Sample observed value:

```
INFO: Average TPS = 212.6126438300756 +/- 51.15100902127161
```

Note the "transactions" here are just transfers. I'm not sure what the expected values are, so I set the asserts in the test to pass based on the sample value above. cc @SkidanovAlex

Reply:
This still feels low to me given that we have now separated network actors from client actor. I think it is because we always send from the same access key on the same account. We should either send from different accounts or different access key on the same account to achieve better results.

Reply:
This test is already using multiple accounts. Each node has 50 accounts they send transactions from in parallel. All accounts use the same access key, but that is only for simplicity of key management in this test and should not introduce any bottleneck as far as I know.

You can see the accounts are assigned to the nodes here: https://github.com/nearprotocol/nearcore/pull/2922/files#diff-66dfef14f75fed19ad824a296a78270fR78-R82

And the accounts are created in the near-ops repo here: https://github.com/nearprotocol/near-ops/blob/master/mocknet.py#L74

And when I examine the transactions in each block I can see that there are transactions from a range of accounts:

```
>>> account_ids = set([t['signer_id'] for t in chunk['transactions']])
>>> len(account_ids)
162
>>> print(sorted(list(account_ids)))
['load_testing_0', 'load_testing_1', 'load_testing_10', 'load_testing_11', 'load_testing_12', 'load_testing_13', 'load_testing_14', 'load_testing_15', 'load_testing_16', 'load_testing_17', 'load_testing_18', 'load_testing_19', 'load_testing_2', 'load_testing_20', 'load_testing_200', 'load_testing_201', 'load_testing_202', 'load_testing_203', 'load_testing_204', 'load_testing_205', 'load_testing_206', 'load_testing_207', 'load_testing_208', 'load_testing_21', 'load_testing_211', 'load_testing_22', 'load_testing_223', 'load_testing_226', 'load_testing_227', 'load_testing_229', 'load_testing_23', 'load_testing_231', 'load_testing_232', 'load_testing_234', 'load_testing_235', 'load_testing_236', 'load_testing_24', 'load_testing_242', 'load_testing_244', 'load_testing_245', 'load_testing_246', 'load_testing_247', 'load_testing_248', 'load_testing_249', 'load_testing_25', 'load_testing_26', 'load_testing_260', 'load_testing_261', 'load_testing_262', 'load_testing_263', 'load_testing_264', 'load_testing_265', 'load_testing_266', 'load_testing_267', 'load_testing_268', 'load_testing_269', 'load_testing_27', 'load_testing_270', 'load_testing_271', 'load_testing_272', 'load_testing_28', 'load_testing_280', 'load_testing_284', 'load_testing_285', 'load_testing_286', 'load_testing_287', 'load_testing_288', 'load_testing_289', 'load_testing_29', 'load_testing_290', 'load_testing_292', 'load_testing_293', 'load_testing_294', 'load_testing_295', 'load_testing_296', 'load_testing_297', 'load_testing_298', 'load_testing_299', 'load_testing_3', 'load_testing_30', 'load_testing_31', 'load_testing_319', 'load_testing_32', 'load_testing_322', 'load_testing_324', 'load_testing_326', 'load_testing_327', 'load_testing_328', 'load_testing_329', 'load_testing_33', 'load_testing_330', 'load_testing_331', 'load_testing_332', 'load_testing_333', 'load_testing_334', 'load_testing_335', 'load_testing_336', 'load_testing_337', 'load_testing_338', 'load_testing_339', 'load_testing_34', 'load_testing_340', 'load_testing_341', 'load_testing_342', 'load_testing_343', 'load_testing_344', 'load_testing_346', 'load_testing_348', 'load_testing_349', 'load_testing_35', 'load_testing_350', 'load_testing_351', 'load_testing_352', 'load_testing_353', 'load_testing_354', 'load_testing_355', 'load_testing_356', 'load_testing_357', 'load_testing_358', 'load_testing_359', 'load_testing_36', 'load_testing_360', 'load_testing_361', 'load_testing_362', 'load_testing_363', 'load_testing_364', 'load_testing_365', 'load_testing_366', 'load_testing_367', 'load_testing_368', 'load_testing_37', 'load_testing_38', 'load_testing_387', 'load_testing_388', 'load_testing_389', 'load_testing_39', 'load_testing_390', 'load_testing_391', 'load_testing_392', 'load_testing_393', 'load_testing_394', 'load_testing_395', 'load_testing_396', 'load_testing_397', 'load_testing_398', 'load_testing_399', 'load_testing_4', 'load_testing_40', 'load_testing_41', 'load_testing_42', 'load_testing_43', 'load_testing_44', 'load_testing_45', 'load_testing_46', 'load_testing_47', 'load_testing_48', 'load_testing_49', 'load_testing_5', 'load_testing_6', 'load_testing_7', 'load_testing_8', 'load_testing_9']
```

Reply:
I see. Does the number improve if you change 50 to 100?

Reply:
Changing the number of accounts to 100 gives a TPS of `350 +/- 70`. Should I try to see how many accounts I can add before the TPS levels out?

Reply:
@birchmd I think you will quickly run into the problem of rpc not being able to take that many requests before reaching the maximum capacity of the network. You can try 200, but I think that will already be problematic. Maybe we should spin up more nodes :)